stl/inc/xloctime

```diff
@@ -554,26 +558,27 @@ private:
             }
         }
 
-        bool _Seendigit = false;
+        int _NSeendigit = 0;
 
         for (; _First != _Last && _Ctype_fac.narrow(*_First) == '0'; ++_First) { // strip leading zeros
-            _Seendigit = true;
+            ++_NSeendigit;
         }
 
-        if (_Seendigit) {
-            *_Ptr++ = '0'; // replace one or more with single zero
+        if (_NSeendigit > 0) {
+            *_Ptr++     = '0'; // replace one or more with single zero
+            _NSeendigit = 1;

```

Comment:
This logic for resetting the number of digits seen when leading zeros are encountered appears to be incorrect - it results in successfully parsing `istringstream iss("202000000000000923");`, but Clang on Compiler Explorer reports `ios_base::failbit` for that. Therefore, I recommend adding that as a test case.

(The existing line "replace one or more with single zero" appears to be fine, as it's used for the actual conversion.)

Reply:
Nice catch. Sorry for missing this case.

Reply:
No problem, that's what code review is for :smile_cat: